src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -627,6 +629,219 @@ class Polyhedron
+  {
+    // Initialize our polyhedron to return
+    Polyhedron<T, NDIMS> poly;
+
+    poly.addVertex(hex[0]);
+    poly.addVertex(hex[1]);
+    poly.addVertex(hex[2]);
+    poly.addVertex(hex[3]);
+    poly.addVertex(hex[4]);
+    poly.addVertex(hex[5]);
+    poly.addVertex(hex[6]);
+    poly.addVertex(hex[7]);
+
+    poly.addNeighbors(0, {1, 4, 3});
+    poly.addNeighbors(1, {0, 2, 5});
+    poly.addNeighbors(2, {1, 3, 6});
+    poly.addNeighbors(3, {2, 0, 7});
+    poly.addNeighbors(4, {0, 5, 7});
+    poly.addNeighbors(5, {1, 6, 4});
+    poly.addNeighbors(6, {2, 7, 5});
+    poly.addNeighbors(7, {3, 4, 6});
+
+    // Reverses order of vertices 1,3 and 5,7 if volume is negative
+    if(checkSign)
+    {
+      if(poly.volume() < 0)
+      {
+        axom::utilities::swap<Point<T, NDIMS>>(poly[1], poly[3]);
+        axom::utilities::swap<Point<T, NDIMS>>(poly[5], poly[7]);
+      }
+    }
+
+    return poly;
+  }
+
+  /*!
+ * \brief Creates a Polyhedron from a given Octahedron's vertices.
+ *
+ * \param [in] oct The octahedron
+ * \param [in] checkSign If true (default is false), checks the volume of the
+ *             Polyhedron is positive. If volume is negative, order of some
+ *             vertices will be swapped.
+ *
+ * \return A Polyhedron with the Octahedron's vertices and added
+ *         vertex neighbors
+ *
+ * \note The Octahedron is assumed to have a specific vertex order:
+ * \verbatim
+ *
+ *            0                +y
+ *            /\                    +z
+ *       4 --/  \-- 5           ^  >
+ *         \/    \ /            | /
+ *         /      \             |/
+ *       2 -------- 1           -----> +x
+ *            \/
+ *            3
+ *
+ * \endverbatim
+ *
+ *       The Polyhedron's vertex neighbors are created assuming this vertex
+ *       ordering.
+ *
+ * \note checkSign flag does not guarantee the Polyhedron's vertex order
+ *       will be valid. It is the responsiblity of the caller to pass
+ *       a Octahedron with a valid vertex order.
+ */
+  AXOM_HOST_DEVICE
+  static Polyhedron from_primitive(const Octahedron<T, NDIMS>& oct,
+                                   bool checkSign = false)
+  {
+    // Initialize our polyhedron to return
+    Polyhedron<T, NDIMS> poly;
+
+    poly.addVertex(oct[0]);
+    poly.addVertex(oct[1]);
+    poly.addVertex(oct[2]);
+    poly.addVertex(oct[3]);
+    poly.addVertex(oct[4]);
+    poly.addVertex(oct[5]);
+
+    poly.addNeighbors(0, {1, 5, 4, 2});
+    poly.addNeighbors(1, {0, 2, 3, 5});
+    poly.addNeighbors(2, {0, 4, 3, 1});
+    poly.addNeighbors(3, {1, 2, 4, 5});
+    poly.addNeighbors(4, {0, 5, 3, 2});
+    poly.addNeighbors(5, {0, 1, 3, 4});
+
+    // Reverses order of vertices 1,2 and 4,5 if volume is negative.
+    // Expanded swap operations for HIP workaround.
+    if(checkSign)
+    {
+      double vol = poly.volume();
+      if(vol < 0)
+      {
+        PointType p2({poly[4][0], poly[4][1], poly[4][2]});
```

Comment:
Do the neighbors not need to be changed for the swapped vertices?

Reply:
> Do the neighbors not need to be changed for the swapped vertices?
Just the vertex order, because the neighbor ordering is pre-set based on the order of vertices.